src/StaticLibrary.cpp

```diff
@@ -0,0 +1,193 @@
+#include "StaticLibrary.h"
+
+#include <stdio.h>
+#ifndef _MSC_VER
+#include <sys/stat.h>
+#include <unistd.h>
+#endif
+
+#include "Error.h"
+
+namespace Halide {
+namespace Internal {
+
+namespace {
+
+class File {
```

Comment:
Agreed. Although Zalman pointed out offline that std::ofstream might be a better choice for some of this.